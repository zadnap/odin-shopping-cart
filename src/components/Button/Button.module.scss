@use '@/styles/variables.scss' as *;

.button {
  color: $text-color;
  background-color: $subbackground-color;
  border: none;
  cursor: pointer;
  border-radius: 10px;
  padding: 10px 20px;

  &.round {
    border-radius: 100px;
  }

  &.square {
    padding: 10px 15px;
  }

  &.active {
    color: $background-color;
    background-color: $text-color;
  }
}
