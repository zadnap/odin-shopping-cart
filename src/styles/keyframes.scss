@keyframes fadeIn {
  from {
    opacity: 0.5;
  }
  to {
    opacity: 1;
  }
}

@keyframes rotate {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@keyframes pulse {
  0% {
    opacity: 1;
    transform: scale(1);
  }
  40% {
    opacity: 0.6;
    transform: scale(0.95);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
}
