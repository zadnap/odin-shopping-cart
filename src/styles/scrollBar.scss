@use './variables.scss' as *;

::-webkit-scrollbar {
  width: 7px;
  height: 7px;
}

::-webkit-scrollbar-track {
  background-color: transparent;
  border-radius: 10px;
}

::-webkit-scrollbar-thumb {
  background-color: $subtext-color;
  border-radius: 10px;
}
