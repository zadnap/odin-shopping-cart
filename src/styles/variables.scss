$background-color: #0e1518;
$subbackground-color: #bbc7cc34;
$text-color: #f9f8ff;
$subtext-color: #959ca3;
$overlay-color: #000000b7;

$border-radius: 5px;
$navigation-width: 250px;
$header-height: 68.5px;
$gap: 15px;

$mobile-max: 425px;
$tablet-max: 768px;
