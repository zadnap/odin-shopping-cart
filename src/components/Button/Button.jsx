import styles from './Button.module.scss';

function Button({
  round,
  square,
  active,
  disabled,
  onClick,
  children,
  className,
  ...props
}) {
  return (
    <button
      className={`
        ${styles.button} 
        ${round ? styles.round : ' '}
        ${square ? styles.square : ' '}
        ${active ? styles.active : ' '}
        ${className}
      `}
      disabled={disabled}
      onClick={onClick}
      {...props}
    >
      {children}
    </button>
  );
}

export default Button;
