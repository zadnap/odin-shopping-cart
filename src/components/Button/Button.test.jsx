import { vi, describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from './Button';
import styles from './Button.module.scss';

describe('Button component', () => {
  it('should render to match snapshot', () => {
    const { container } = render(<Button>Click Me</Button>);

    expect(container).toMatchSnapshot();
  });

  it('should apply round class when round prop is true', () => {
    render(<Button round>Round Button</Button>);
    const button = screen.getByRole('button');

    expect(button.classList.contains(styles.round)).toBe(true);
  });

  it('should apply square class when square prop is true', () => {
    render(<Button square>Square Button</Button>);
    const button = screen.getByRole('button');

    expect(button.classList.contains(styles.square)).toBe(true);
  });

  it('should call onClick handler when clicked', async () => {
    const onClick = vi.fn();
    const user = userEvent.setup();
    render(<Button onClick={onClick} />);

    const button = screen.getByRole('button');

    await user.click(button);

    expect(onClick).toHaveBeenCalled();
  });
});
