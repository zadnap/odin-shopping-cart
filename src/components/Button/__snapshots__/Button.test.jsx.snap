// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > should render to match snapshot 1`] = `
<div>
  <button
    class="_button_ee73f7  "
  >
    Click Me
  </button>
</div>
`;
